### PR TITLE
MAINT: Bump versions and build config to 25.01 with RAPIDS 24.10

### PR DESCRIPTION
**Pull request by seberg**
The first two commits just update all versions and modify the build config the same way as for `legate-dataframe`.

The last commit has a few miscellaneous build related changes (mainly to `build.sh` because `./build.sh clean` stopped working).

The build seems weird/messy to me, but I don't think that _changed_.  My guess is, that the Python/C++ builds should be seperated out into two `CMakeLists.txt` (similiar to legate-df).

**NOTE: Yes, this includes changing `VERSION`, so I'll just plan on aborting the actions run to avoid uploading another `24.11.00` experimental release before tagging `25.01.00.dev`.**

### CODE REVIEW

**Comment by seberg** [comment]
We also have a few new build warnings due to `kmeans` moving to `cuvs`, so that will need to be fixed eventually.

File: pyproject.toml [hunk 1]
```diff
@@ -36,29 +36,49 @@ License = "https://github.com/rapidsai/legate-raft/blob/main/LICENSE"
...
+regex = "(?P<value>.*)"
+
+[tool.scikit-build]
+build-dir = "build/{wheel_tag}"
+cmake.build-type = "Release"
+cmake.version = "CMakeLists.txt"
+minimum-version = "build-system.requires"
+ninja.make-fallback = true
+sdist.exclude = ["*tests*"]
+sdist.include = ["legate_raft/install_info.py"]
```
Comment (seberg) [inline pyproject.toml:79]: `scikit-build-core` honors `.gitignore`, took me a while to realize why this was missing.

**Comment by seberg** [comment]
Yeah, the build stayed about as slow as it was (maybe a tiny bit slower), but it definitely is building things twice unnecessarily (and always was).
We should figure this out (although I might prefer to not do it here).

File: pyproject.toml (hunk 1 above)
Comment (jameslamb) [reply pyproject.toml:79]: oh wow! I actually did not know that, thanks for the comment.

**Review by jameslamb — APPROVED** [review]
Looks good!

I agree, the duplicate build could be removed in a later PR. In `legate-boost`, I did that by requiring an argument for every type of build. Like this:

https://github.com/rapidsai/legate-boost/blob/4f72e1b7f20cdf1d3b27fa129ec82cb06216cd18/build.sh#L52

So there you have like:

```shell
# build with 'cmake --build'
./build.sh liblegateboost

# build with 'pip install'
./build.sh legate-boost
```

And then the conda recipe runs:

```shell
./build.sh legate-boost
```

https://github.com/rapidsai/legate-boost/blob/4f72e1b7f20cdf1d3b27fa129ec82cb06216cd18/conda/recipes/legate-boost/build.sh#L4

**Comment by seberg** [comment]
> did that by requiring an argument for every type of build. Like this:

Thanks for the tip.  That is definitely quicker than trying to split things up.

**Pull request merged**